python/common/org/python/types/Generator.java

```diff
@@ -26,7 +30,12 @@ public Generator(
         this.expression = expression;
         this.yield_point = 0;
         this.stack = stack;
-        this.message = new org.python.types.NoneType();
+        this.message = org.python.types.NoneType.NONE;
```

Comment:
We're acquiring a lot of "state" variables in the Generator; I can see that they're all needed (with the possible exception of just_started), but it would be helpful to have some explanatory notes of what the values all mean in the generator context. (e.g., `expression == null` means the generator has exhausted itself). A class-level documentation block explaining all the possible "states" of a generator is all we'd need here.

Reply:
Resolved. Added a class-level documentation block explaining possible states of a generator.